Source/WebGPU/WebGPU/Buffer.h

```diff
@@ -148,6 +151,17 @@ class Buffer : public WGPUBufferImpl, public RefCounted<Buffer>, public CanMakeW
 #if CPU(X86_64)
     bool m_mappedAtCreation { false };
 #endif
-};
+} SWIFT_SHARED_REFERENCE(retainBuffer, releaseBuffer);
 
 } // namespace WebGPU
+
+inline void retainBuffer(WebGPU::Buffer* obj)
+{
+    WTF::RetainReleaseHelperRefCounted::retain(obj);
```

Comment:
What prevents us from just calling obj->ref() here?

I believe we can remove the RetainReleaseHelper classes. If we can static_assert that T is derived from RefCounted, that means the compiler can see that T derives from RefCounted at compile time, so RefCounted::ref() should be in scope.

(Apologies if I'm repeating this comment. I feel like I said it before, but I don't see it anywhere.)

Reply:
This is basically a placeholder for if we want to do something custom for retain release. Like logging or some workarounds.
Right now we have 2 classes getting imported but that number will go up significantly.
All of the quirks can just be in the helper class and not at callsites

Reply:
Do we need these manual definitions in each header file?

Couldn't it be directly integrated into RefCounted.h for instance?
```
template<typename T>
static void retainRefCounted(T* obj)
{
...
}
template<typename T>
static void releaseRefCounted(T* obj)
{
...
}
```

that way we don't need manual definitions in each header file?